cmd/starbridge/model/chain.go

```diff
@@ -34,10 +42,32 @@ var (
 			AssetStellar_WUSDC.MapKey(): AssetEthereum_USDC,
 			AssetStellar_XLM.MapKey():   AssetEthereum_WXLM,
 		},
-		nextNonceFn: unsupportedNonceForChain, // TODO we haven't had the time to add the logic to go from Stellar to Ethereum yet
+		nextNonceFn:                  unsupportedNonceForChain, // TODO NS we haven't had the time to add the logic to go from Stellar to Ethereum yet
+		ValidateDestinationAddressFn: unsupportedValidateDestinationAddressFn,
 	}
 )
 
+func computeAllAssetMap(chain *Chain) map[string]*AssetInfo {
+	m := map[string]*AssetInfo{}
+	for _, v := range chain.AddressMappings {
+		m[v.MapKey()] = v
+	}
+
+	// native asset is included in the above list since it will be in the AddressMappings but add explicitly here too
+	m[chain.NativeAsset.MapKey()] = chain.NativeAsset
+
+	logger.Debugf("added %d items when creating AllAssetMap for chain=%s", len(m), chain.Name)
+	return m
+}
+
+func init() {
+	logger = supportlog.New()
+	logger.SetLevel(logrus.InfoLevel)
+
+	ChainStellar.AllAssetMap = computeAllAssetMap(ChainStellar)
+	ChainEthereum.AllAssetMap = computeAllAssetMap(ChainEthereum)
```

Comment:
💡 Can we put these calls in the definition of `ChainStellar` and `ChainEthereum`? I don't think they need to live separately in an `init` function.

Reply:
I didn't want to repeat having to pass in the assets. I will be creating a factory method for the instantiation of the Chain and this would be done automatically inside the factory method. Thoughts on that approach to avoid repetition but also avoid an `init` function?

Reply:
Ah, I missed that the functions accepted a reference to itself. Got it.